python/dgl/sparse.py

```diff
@@ -366,124 +366,75 @@ def _bwd_segment_cmp(feat, arg, m):
                                  to_dgl_nd_for_write(out))
     return out
 
-class CSRMatrix(object):
-    """Device- and backend-agnostic sparse matrix in CSR format.
+def csrmm(A, A_weights, B, B_weights, num_vtypes):
+    """Return a graph whose adjacency matrix is the sparse matrix multiplication
+    of those of two given graphs.
 
     Parameters
     ----------
-    data : Tensor
-        The data array.
-    indices : Tensor
-        The column indices array.
-    indptr : Tensor
-        The row index pointer array.
-    num_rows : int
-        The number of rows.
-    num_cols : int
-        The number of columns.
-    """
-    def __init__(self, data, indices, indptr, num_rows, num_cols):
-        self.indptr = indptr
-        self.indices = indices
-        self.data = data
-        self.shape = (num_rows, num_cols)
-
-def csrmm(A, B):
-    """Sparse-sparse matrix multiplication.
-
-    This is an internal function whose interface is subject to changes.
-
-    Parameters
-    ----------
-    A : dgl.sparse.CSRMatrix
-        The left operand
-    B : dgl.sparse.CSRMatrix
-        The right operand
+    A : HeteroGraphIndex
+        The input graph index as left operand.
+    A_weights : Tensor
+        The edge weights of graph A as 1D tensor.
+    B : HeteroGraphIndex
+        The input graph index as right operand.
+    B_weights : Tensor
+        The edge weights of graph B as 1D tensor.
+    num_vtypes : int
+        The number of node types for the returned graph (must be either 1 or 2).
```

Comment:
No we can't.  For instance multiplying a graph with metagraph `(A, AB, B)` and a graph with `(B, BA, A)` should give the resulting graph `(A, ..., A)` with one node type.

Reply:
Can we infer the number of node types based on the input without the need of user specification?

Reply:
We can infer the number of node types from the metagraphs, but only in the Python because we need to inspect the node type names.  This function is just a bare Python wrapper of the C function so I'm not doing it here.  For the corresponding user interface we won't require the user to specify the number of node types.